src/machine/image-dbus.c

```diff
@@ -56,11 +61,37 @@ int bus_image_method_remove(
         if (r == 0)
                 return 1; /* Will call us back */
 
-        r = image_remove(image);
-        if (r < 0)
+        if (pipe2(errno_pipe_fd, O_CLOEXEC|O_NONBLOCK) < 0)
+                return sd_bus_error_set_errnof(error, errno, "Failed to create pipe: %m");
+
```

Comment:
this is a bit much copy-pasta from https://github.com/systemd/systemd/pull/3153/commits/3c7e249d30fb034423aad9361c9bfe5d7dc0ff63

Why not have the operations interface implement the fork/pipe/etc and just have an operation_run() function you supply the closure to run asynchronously (image_remove() vs. image_clone())?